src/engine/strat_engine/filesystem.rs

```diff
@@ -64,13 +64,14 @@ impl StratFilesystem {
     pub fn snapshot(&self,
                     dm: &DM,
                     thin_pool: &ThinPoolDev,
-                    snapshot_name: &DmName,
+                    snapshot_name: &str,
```

Comment:
This method definitely needs a header comment, what with its expanded list of parameters.

Reply:
Ok... updated to:

    /// Create a snapshot of the filesystem. Return the resulting filesystem/ThinDev
    /// to the caller.  Use snapshot_name for the Stratis filesytem name.  Use
    /// snapshot_dmname for the new name of the ThinDev allocated for the snapshot.
    /// Mounting a filesytem with a duplicate UUID would require special handling,
    /// so snapshot_fs_uuid is used to update the new snapshot filesystem so it has
    /// a unique UUID.